apps/sr/tid1500writer.cxx

```diff
@@ -105,15 +98,21 @@ int main(int argc, char** argv){
       cout << "Loading " << dicomFilePath << endl;
       CHECK_COND(ff.loadFile(dicomFilePath.c_str()));
 
-      DcmDataset imageLibDataset;
-      DcmTag tagsToCopy[] = {DCM_SOPClassUID,DCM_SOPInstanceUID,DCM_Modality,DCM_StudyDate,DCM_Columns,DCM_Rows,DCM_PixelSpacing,DCM_BodyPartExamined,DCM_ImageOrientationPatient,DCM_ImagePositionPatient};
-      for(int t=0;t<10;t++){
-        copyElement(tagsToCopy[t], ff.getDataset(), &imageLibDataset);
+
+      if(i==0){
+        DcmDataset imageLibGroupDataset;
+        DcmTag commonTagsToCopy[] =   {DCM_SOPClassUID,DCM_Modality,DCM_StudyDate,DCM_Columns,DCM_Rows,DCM_PixelSpacing,DCM_BodyPartExamined,DCM_ImageOrientationPatient};
+        for(int t=0;t<STATIC_ARRAY_SIZE(commonTagsToCopy);t++){
+          ff.getDataset()->findAndInsertCopyOfElement(commonTagsToCopy[t],&imageLibGroupDataset);
+        }
+        CHECK_COND(report.getImageLibrary().addImageEntryDescriptors(imageLibGroupDataset));
       }
 
-      OFCHECK(report.getImageLibrary().addImageGroup().good());
-      OFCHECK(report.getImageLibrary().addImageEntry(imageLibDataset).good());
-      OFCHECK(report.getImageLibrary().addImageEntryDescriptors(imageLibDataset).good());
+      DcmDataset imageEntryDataset;
+      DcmTag imageTagsToCopy[] = {DCM_Modality,DCM_SOPClassUID,DCM_SOPInstanceUID,DCM_ImagePositionPatient};
```

Comment:
Two further comments: Instead of gotoChild() I would suggest to go to the "Modality" child content item using gotoNamedChildNode(), which is more robust IMHO. Also the code duplication (calling gotoChild() and removeSubTree() twice, before and within the while loop) could be avoided but is probably OK for a temporary workaround like this.